.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,39 @@
+name: Clang Format Check
+
+on:
+  pull_request:
+    paths:
+      - 'src/**'
+
+jobs:
+  clang-format-check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Set up Clang
+      run: |
+        sudo apt-get update -y
+        sudo apt-get upgrade -y
+        sudo apt-get install -y clang-format
+    - name: Run clang-format
+      id: clang-format
+      run: |
+        # Run clang-format and capture the diff
+        cd src
+        clang-format -i **/*.c **/*.h
+        # Capture the diff output
+        DIFF=$(git diff)
+        if [ ! -z "$DIFF" ]; then
+          echo "::set-output name=diff::$DIFF"
```

Comment:
"The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. .."